### PR TITLE
Switch to `mamba` for dependency management, and document other required installation details

### PR DESCRIPTION
**Pull request by amorehead**
* Transitions the project to instead use `mamba` for dependency management, as `mamba` in my experience is oftentimes more platform-portable than Singularity/Apptainer.
* Documents other required installation details including `make_ss.sh` and `$PIPE_DIR` references.
* Fixes merging of MSAs following https://github.com/baker-laboratory/RoseTTAFold-All-Atom/pull/11.

### CODE REVIEW

**Review by r-krishna — CHANGES_REQUESTED** [review]
Thank you for creating the mamba dependencies for this project! I think users will really benefit from having this! 
Two minor points: 
1. I will upload the protein_sm example. My apologies for this. I would prefer to keep the current README example since that example was shown in the ms. 
2. I am not able to reproduce the bug that requires changing a3m["msa"][ to a3m[0], if you could send that to me I'm happy to debug!

File: rf2aa/data/merge_inputs.py [hunk 1]
```diff
@@ -73,8 +73,8 @@ def merge_protein_inputs(protein_inputs, deterministic: bool = False):
         chain_lengths = list(zip(protein_inputs.keys(), lengths_list))
 
         merged_input = RawInputData(
-            a3m_out["msa"],
-            a3m_out["ins"],
+            a3m_out[0],
```
[the remark this answers is not in the document]
Comment (r-krishna) [reply rf2aa/data/merge_inputs.py:76]: Ok i was able to find the problem here and will fix it imminently. Please revert this commit and I will merge this PR in! Thank you so much for your contribution

File: README.md [hunk 2]
```diff
@@ -118,32 +140,33 @@ This repo currently does not support making RNA MSAs or pairing protein MSAs wit
...
 Now, predict the example protein/NA complex. 
 ```
-SE3nv-20240131.sif -m rf2aa.run_inference --config-name nucleic_acid
+python -m rf2aa.run_inference --config-name nucleic_acid
 ```
 <a id="p-sm-complex"></a>
 ### Predicting Protein Small Molecule Complexes
 To predict protein small molecule complexes, the syntax to input the protein remains the same. Adding in the small molecule works similarly to other inputs. 
-Here is an example (from `rf2aa/config/inference/protein_sm.yaml`):
+Here is an example (from `rf2aa/config/inference/protein_complex_sm.yaml`):
```
[the remark this answers is not in the document]
Comment (amorehead) [reply README.md]: No problem. I've reverted the README to only reference `protein_sm.yaml` in [this commit](https://github.com/baker-laboratory/RoseTTAFold-All-Atom/pull/13/commits/ec93ac6346ad408105d1b7ef638245b35dd9602c).

File: rf2aa/data/merge_inputs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (amorehead) [reply rf2aa/data/merge_inputs.py]: Sounds good. I've reverted this change in [this commit](https://github.com/baker-laboratory/RoseTTAFold-All-Atom/pull/13/commits/1fcd8ec1a95a1e99ec16a560ef51a22f0b1a21a2).

**Pull request merged**